example/BatchProducer.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <condition_variable>
+#include <iomanip>
+#include <iomanip>
+#include <iostream>
+#include <iostream>
+#include <mutex>
+#include <thread>
+#include <vector>
+#include "common.h"
+
+using namespace rocketmq;
+using namespace std;
+boost::atomic<bool> g_quit;
+std::mutex g_mtx;
+std::condition_variable g_finished;
+TpsReportService g_tps;
+
+void SyncProducerWorker(RocketmqSendAndConsumerArgs* info,
+                        DefaultMQProducer* producer) {
+    while (!g_quit.load()) {
+        if (g_msgCount.load() <= 0) {
+            std::unique_lock<std::mutex> lck(g_mtx);
+            g_finished.notify_one();
+            break;
+        }
+
+        vector<MQMessage> msgs;
+        MQMessage msg1(info->topic, "*", info->body);
+        msg1.setProperty("property1", "value1");
+        MQMessage msg2(info->topic, "*", info->body);
+        msg2.setProperty("property1", "value1");
+        msg2.setProperty("property2", "value2");
+        MQMessage msg3(info->topic, "*", info->body);
+        msg3.setProperty("property1", "value1");
+        msg3.setProperty("property2", "value2");
+        msg3.setProperty("property3", "value3");
+        msgs.push_back(msg1);
+        msgs.push_back(msg2);
+        msgs.push_back(msg3);
+        try {
+            auto start = std::chrono::system_clock::now();
+            SendResult sendResult = producer->send(msgs);
+            g_tps.Increment();
+            --g_msgCount;
+            auto end = std::chrono::system_clock::now();
+            auto duration =
+                    std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+            if (duration.count() >= 500) {
+                std::cout << "send RT more than: " << duration.count()
```

Comment:
use log instead of std::cout

Reply:
i suggest use std::cout or printf for samples because it can see the output in screen at once